src/replication.c

```diff
@@ -2503,7 +2503,8 @@ int slaveTryPartialResynchronization(connection *conn, int read_reply) {
      * return PSYNC_TRY_LATER if we believe this is a transient error. */
 
     if (!strncmp(reply,"-NOMASTERLINK",13) ||
-        !strncmp(reply,"-LOADING",8))
+        !strncmp(reply,"-LOADING",8) ||
+        !strncmp(reply,"-Reading from master",20))
```

Comment:
i don't like this string compare to a fake error string we just made up.
it was arguably ok if the caller handled it the same as any other error, but if we want to handle it differently, let's return an better error indication. maybe NULL?
we can then still log errno outside, or even inside receiveSynchronousResponse?

Reply:
I think it is very useful to log this error explicitly for debugging purposes. Currently the error `Unexpected reply to PSYNC from master: -Reading from master: Input/output error` is logged in the calling function `slaveTryPartialResynchronization`. 

Maybe I can log this error with errno inside `receiveSynchronousResponse` and return NULL.